src/main/java/cmu/pasta/mu2/diff/plugin/DiffGoal.java

```diff
@@ -188,6 +203,77 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             throw new MojoExecutionException("Invalid Mutation OptLevel!");
         }
 
+        // Set filters to default value of empty string
+        if (filters == null){
+            filters = "";
+        }
+
+        // Initialize empty list of filters
+        List<MutantFilter> filterList = new ArrayList<>();
+        int i = 0;
+        char[] filterChars = filters.toCharArray();
+        String current = "";
+        Filter f;
+        int len = filterChars.length;
+
+        // Parse input string (of format "filter1:arg1,arg2,...,filter2:arg1,arg2,...") to create list of filters
+        while(i < len){
+            char c = filterChars[i];
+            if (c == ':') {
+                try {
+                    f = Filter.valueOf(current.toUpperCase());
+                } catch (IllegalArgumentException e) {
+                    throw new MojoExecutionException("Invalid Mutation Filter!");
+                }
+                if (f == Filter.KRANDOM){
+                    
+                    i++;
+                    String arg = "";
+                    while (i < len && filterChars[i] != ','){
+                        arg += filterChars[i];
+                        i ++;
+                    }
+                    try {
+                        filterList.add(new KRandomFilter(Integer.parseInt(arg)));
+                    } catch (NumberFormatException e) {
+                        throw new MojoExecutionException("Invalid Mutation Filter Argument(s)!");
+                    }
+
+                } else if (f == Filter.KLEASTEXEC){
+
+                    i++;
+                    String arg = "";
+                    while (i < len && filterChars[i] != ','){
+                        arg += filterChars[i];
+                        i ++;
+                    }
+                    try {
+                        filterList.add(new KLeastExecutedFilter(Integer.parseInt(arg)));
+                    } catch (NumberFormatException e) {
+                        throw new MojoExecutionException("Invalid Mutation Filter Argument(s)!");
+                    }
+
+                } else if (f == Filter.FILE) {
+
+                    i++;
+                    String arg = "";
+                    while (i < len && filterChars[i] != ','){
+                        arg += filterChars[i];
+                        i ++;
+                    }
+                    try {
+                        filterList.add(new FileMutantFilter(arg));
+                    } catch (FileNotFoundException e) {
+                        throw new MojoExecutionException("Invalid Mutation Filter Argument(s)!");
+                    }
+                }
+                current = "";
+            } else {
+                current += c;
+            }
+            i += 1;
+        }
```

Comment:
Can this be factored out into a private method `List<Filter> parseFilters(String filtersSpec)` so that the logic doesn't take up most of the `execute()` method?

Reply:
Pushed those changes!